l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -134,139 +149,206 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
+            factoryDeps: new bytes[](0),
+            refundRecipient: _refundRecipient
+        });
+        l2TxHash = BRIDGE_HUB.requestL2TransactionDirect{value: msg.value}(request);
+    }
+
     /// @notice Allows bridgehub to acquire mintValue for L1->L2 transactions.
     /// @dev If the corresponding L2 transaction fails, refunds are issued to a refund recipient on L2.
-    /// @param _chainId The chain ID of the hyperchain to which deposit.
+    /// @param _chainId The chain ID of the ZK chain to which deposit.
+    /// @param _assetId The deposited asset ID.
     /// @param _prevMsgSender The `msg.sender` address from the external call that initiated current one.
-    /// @param _l1Token The L1 token address which is deposited.
     /// @param _amount The total amount of tokens to be bridged.
     function bridgehubDepositBaseToken(
         uint256 _chainId,
+        bytes32 _assetId,
         address _prevMsgSender,
-        address _l1Token,
         uint256 _amount
     ) external payable virtual onlyBridgehubOrEra(_chainId) whenNotPaused {
-        if (_l1Token == ETH_TOKEN_ADDRESS) {
-            require(msg.value == _amount, "L1SharedBridge: msg.value not equal to amount");
-        } else {
-            // The Bridgehub also checks this, but we want to be sure
-            require(msg.value == 0, "ShB m.v > 0 b d.it");
+        (address l1AssetHandler, bytes32 assetId) = _getAssetProperties(_assetId);
+        _transferAllowanceToNTV(assetId, _amount, _prevMsgSender);
+        // slither-disable-next-line unused-return
+        IL1AssetHandler(l1AssetHandler).bridgeBurn{value: msg.value}({
+            _chainId: _chainId,
+            _mintValue: _amount,
+            _assetId: assetId,
+            _prevMsgSender: _prevMsgSender,
+            _data: abi.encode(_amount, address(0))
+        });
 
-            uint256 amount = _depositFunds(_prevMsgSender, IERC20(_l1Token), _amount); // note if _prevMsgSender is this contract, this will return 0. This does not happen.
-            require(amount == _amount, "3T"); // The token has non-standard transfer logic
+        // Note that we don't save the deposited amount, as this is for the base token, which gets sent to the refundRecipient if the tx fails
+        emit BridgehubDepositBaseTokenInitiated(_chainId, _prevMsgSender, _assetId, _amount);
+    }
+
+    /// @notice Returns the address of asset handler and parsed assetId, if padded token address was passed
+    /// @dev For backwards compatibility we pad the l1Token to become a bytes32 assetId.
+    /// @dev We deal with this case here. We also register the asset.
+    /// @param _assetId bytes32 encoding of asset Id or padded address of the token
+    function _getAssetProperties(bytes32 _assetId) internal returns (address l1AssetHandler, bytes32 assetId) {
+        // Check if the passed id is the address and assume NTV for the case
+        assetId = uint256(_assetId) <= type(uint160).max
```

Comment:
This logic is redundant. Please make sure it is never the case in smart contracts. Increasing complexity in smart contracts cost longer internal review, worst external review quality and harder maintaining of the code. 

Now for this specific case.
- `_getAssetProperties` is only called once in `bridgehubDepositBaseToken`
- `bridgehubDepositBaseToken` can be called by:
    - `requestL2Transaction`, the token asset id is hardcoded
    - `requestL2TransactionDirect`, token asset ID is either zero or actual calculated hash:`baseTokenAssetId[_request.chainId]`
    - `requestL2TransactionTwoBridges`, the same as `requestL2TransactionDirect`.

Considering all above, this logic can be completely removed, the function should only return `l1AssetHandler`